### ISSUE TITLE
_openssl.so: undefined symbol: X509_VERIFY_PARAM_set_hostflags on Debian 7.9

### ISSUE THREAD

**Issue by Crashdoom** [certbot/certbot#1335]
I'm following the Quick Start guide and have come across the following error on Debian 7.9, but can't seem to find any mention of this or a resolution:

````
crashdoom@apollo:~$ cd /git/letsencrypt/
crashdoom@apollo:/git/letsencrypt$ ./letsencrypt-auto --agree-dev-preview --server https://acme-v01.api.letsencrypt.org/directory auth
Updating letsencrypt and virtual environment dependencies......
Running with virtualenv: sudo /home/crashdoom/.local/share/letsencrypt/bin/letsencrypt --agree-dev-preview --server https://acme-v01.api.letsencrypt.org/directory auth
[sudo] password for crashdoom: 
Traceback (most recent call last):
  File "/home/crashdoom/.local/share/letsencrypt/bin/letsencrypt", line 7, in <module>
    from letsencrypt.cli import main
  File "/home/crashdoom/.local/share/letsencrypt/local/lib/python2.7/site-packages/letsencrypt/cli.py", line 15, in <module>
    import OpenSSL
  File "/home/crashdoom/.local/share/letsencrypt/local/lib/python2.7/site-packages/OpenSSL/__init__.py", line 8, in <module>
    from OpenSSL import rand, crypto, SSL
  File "/home/crashdoom/.local/share/letsencrypt/local/lib/python2.7/site-packages/OpenSSL/rand.py", line 11, in <module>
    from OpenSSL._util import (
  File "/home/crashdoom/.local/share/letsencrypt/local/lib/python2.7/site-packages/OpenSSL/_util.py", line 6, in <module>
    from cryptography.hazmat.bindings.openssl.binding import Binding
  File "/home/crashdoom/.local/share/letsencrypt/local/lib/python2.7/site-packages/cryptography/hazmat/bindings/openssl/binding.py", line 13, in <module>
    from cryptography.hazmat.bindings._openssl import ffi, lib
ImportError: /home/crashdoom/.local/share/letsencrypt/local/lib/python2.7/site-packages/cryptography/hazmat/bindings/_openssl.so: undefined symbol: X509_VERIFY_PARAM_set_hostflags
crashdoom@apollo:/git/letsencrypt$ uname -a
Linux apollo 2.6.32-39-pve #1 SMP Fri May 8 11:27:35 CEST 2015 i686 GNU/Linux
crashdoom@apollo:/git/letsencrypt$ python -V
Python 2.7.3
````

**Comment by kuba** [comment]
@alex, could you help me here a bit?

**Comment by alex** [comment]
cc: @reaperhulk @dstufft

Hmm, can you `ldd /home/crashdoom/.local/share/letsencrypt/local/lib/python2.7/site-packages/cryptography/hazmat/bindings/_openssl.so` and also `openssl version`?

**Comment by Crashdoom** [comment]
@alex
````
crashdoom@apollo:/git/letsencrypt$ ldd /home/crashdoom/.local/share/letsencrypt/local/lib/python2.7/site-packages/cryptography/hazmat/bindings/_openssl.so 
        linux-gate.so.1 (0xb7777000)
        libssl.so.1.0.0 => /usr/lib/i386-linux-gnu/i686/cmov/libssl.so.1.0.0 (0xb764e000)
        libcrypto.so.1.0.0 => /usr/lib/i386-linux-gnu/i686/cmov/libcrypto.so.1.0.0 (0xb747e000)
        libpthread.so.0 => /lib/i386-linux-gnu/libpthread.so.0 (0xb7462000)
        libc.so.6 => /lib/i386-linux-gnu/libc.so.6 (0xb72f0000)
        libdl.so.2 => /lib/i386-linux-gnu/libdl.so.2 (0xb72eb000)
        /lib/ld-linux.so.2 (0xb7778000)
crashdoom@apollo:/git/letsencrypt$ openssl version
OpenSSL 1.0.1e 11 Feb 2013 (Library: OpenSSL 1.0.2-beta1 24 Feb 2014)
````

**Comment by reaperhulk** [comment]
OpenSSL 1.0.2-beta1 is likely the issue here. `X509_VERIFY_PARAM_set_hostflags` was added in beta2, but we do version checking via the `OPENSSL_VERSION_NUMBER` checking to see if it is >= `0x10002000L`. This is a slight bug, as we should really use `0x10002002L` (which is beta2 of 1.0.2). I'll put in a PR to fix that, but I'd also strongly recommend against using a beta OpenSSL library if possible (is Debian shipping that for 7.x?).

**Comment by reaperhulk** [comment]
This has now been released in 1.1.1.

**Comment by bmw** [comment]
Closing because the issue was fixed in `cryptography`.
